java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistFieldDeclaration.java

```diff
@@ -41,9 +44,16 @@ public JavassistFieldDeclaration(CtField ctField, TypeSolver typeSolver) {
     @Override
     public Type getType() {
         try {
-            return JavassistFactory.typeUsageFor(ctField.getType(), typeSolver);
+            if (ctField.getGenericSignature() != null && declaringType() instanceof TypeParametrizable) {
```

Comment:
Why do we need the second condition? Can a field be declared in a type that is not parametrizable (an enum maybe?) If that is the case the field could not be generic, right?

Reply:
It's just to make sure that the following cast would work. I like to do the instanceof test before actually casting to a different type. So I don't want this code to break if someone changes something in the rest of the code base.

Reply:
Ok, I think I should change declaringType to return a ReferenceTypeDeclaration and then remove this check. For now I will merge this as it is.